### PR TITLE
1.7.0

### PR DESCRIPTION
**Pull request by ZarlengoToptal**

### CODE REVIEW

**Review by bstellshippo — APPROVED** [review]

File: example.js [hunk 1]
```diff
@@ -121,6 +122,9 @@ shippo.customsdeclaration.create({
 	console.log("rates : %s", JSON.stringify(rates, null, 4));
 	// Get the first rate in the rates results for demo purposes.
 	rate = rates.results[0];
+	if (rates.results[0]) {
+		carrier_account = rates.results[0].carrier_account;
```
Comment (sam-allen-shippo) [inline example.js:126]: is there any chance `rates.results[0].carrier_account` could be null or undefined?

File: example.js [hunk 2]
```diff
@@ -137,4 +141,25 @@ shippo.customsdeclaration.create({
 		//Deal with an error with the transaction
 		console.log("Message: %s", JSON.stringify(transaction.messages, null, 2));
 	}
+	var requested_start_time = new Date();
+	requested_start_time.setDate(requested_start_time.getDate() + 1);
+	var requested_end_time = new Date();
+	requested_end_time.setDate(requested_start_time.getDate() + 1.1);
```
Comment (sam-allen-shippo) [inline example.js:147]: what does adding 1.1 to a Date do?

File: example.js (hunk 1 above)
Comment (ZarlengoToptal) [reply example.js]: There should always be a carrier account for each shipment's rate

File: example.js (hunk 2 above)
Comment (ZarlengoToptal) [reply example.js]: This gives a window for a pick up:
   start: tomorrow at current time -> tomorrow @ 12:00
   end: tomorrow at current time + 0.1 days (2.4 hrs) -> tomorrow @ 14:24

File: example_batch.js [hunk 3]
```diff
@@ -50,8 +50,8 @@ var myBatch = {
           "distance_unit": "in",
           "weight": "2",
           "mass_unit": "oz"
-        }]
-      }
+        }}
```
Comment (sam-allen-shippo) [inline example_batch.js:53]: It's a little hard to figure this out just based on the diff - what's the effect of changing the brackets here?

File: example_tracking.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ a tracking number as well as how to register a webhook for tracking updates
 var shippo = require('shippo')('<YOUR_PRIVATE_KEY>');
 
 // example Tracking object for tracking a shipment
-shippo.track.get_status('usps', '1122334455667788')
+shippo.track.get_status('shippo', 'SHIPPO_TRANSIT')
```
Comment (sam-allen-shippo) [inline example_tracking.js:11]: what's the rationale behind changing the example? I think there's some value in having example values that resemble real values

File: example_batch.js (hunk 3 above)
Comment (ZarlengoToptal) [reply example_batch.js]: It was missing square bracket above, has been corrected to create an array

File: CHANGELOG [hunk 5]
```diff
@@ -1,3 +1,6 @@
+1.7.0 Oct 4, 2021
```
Comment (sam-allen-shippo) [inline CHANGELOG:1]: can you update this to Jan 2022? :)

File: example.js [hunk 6]
```diff
@@ -119,8 +121,18 @@ shippo.customsdeclaration.create({
 	process.exit(1);
 }).then(function(rates) {
 	console.log("rates : %s", JSON.stringify(rates, null, 4));
-	// Get the first rate in the rates results for demo purposes.
-	rate = rates.results[0];
+	try {
+		const filteredRates = rates.results.filter(rate => rate.provider.toUpperCase().includes('USPS') || rate.provider.toUpperCase().includes('DHL Express'));
```
Comment (sam-allen-shippo) [inline example.js:125]: The toUpperCase() won't work for `DHL Express`

Comment (ZarlengoToptal) [reply example.js]: Corrected

**Review by sam-allen-shippo — APPROVED** [review]

**Pull request merged**